specs/spec-1-kudos.md

```diff
@@ -1,106 +1,310 @@
+**Visitor** no credentials, not logged in.
 
-TODO: finalize the categories
+- View Kudos List
+- View Kudos Detail
+- Share Kudos (list and deep link to a Kudo detail)
 
-### Spam protection
+**User** logged in to NS.
+
+- Everything visitor can do
+- Comment on Kudos
+- React to Kudos 
+
+**Verified Human** logged in user who has credentials issued by `i-am-human`.
 
-Using proof of personhood is not enough to protect a system from abuse. One user could attach a bot to his account and send thousands of kudos to other accounts (potentially to the same human account).
-We propose to limit that activity: Any human can send maximum 14 kudos per week. The quota will reset every Sunday at midday UTC (12:00). The quota will be a parameter and we will be able to change it in the future.
+- Everything User can do
+- Create new Kudos
+- Recieve new Kudos
 
-Moreover, user should be in control of his Kudos. He will have a full right of removing an unwanted Kudo.
+**Administrator** can take additional actions not availalbe to other actors.
 
-Kudo sender will have to set an expire time. To make it more useful, the contract will require that a minimum expire time is 1month.
+- Everything Verified Human can do
+- Delete Kudos
+- Change Settings
+  - Create or remove categories (aka tags)
+  - Adjust experiation time of Kudos
+  - Delete Kudos
 
-### Smart contract management
+## User Stories / Use Cases
 
-Kudo contract will define a set of administrators which will be able to:
+Here are some user stories or use cases for the Kudos widget based on the provided text:
 
-- create or remove categories
-- adjust min expire time
+1. As a **visitor** (not logged in), I want to view the Kudos list, so I can see the various accomplishments and recognitions in the community.
+2. As a **visitor**, I want to view Kudos details, so I can learn more about a specific Kudo and understand its context.
+3. As a **visitor**, I want to share Kudos (list and deep link to a Kudo detail), so I can promote achievements and encourage others to engage with the community.
+4. As a **logged-in user**, I want to comment on Kudos, so I can express my thoughts and engage with the community.
+5. As a **logged-in user**, I want to react to Kudos, so I can show my appreciation and support for the accomplishments.
+6. As a **verified human**, I want to create new Kudos, so I can recognize and celebrate the contributions and achievements of others in the community.
+7. As a **verified human**, I want to receive new Kudos, so I can be acknowledged for my contributions and achievements.
+8. As an **administrator**, I want to delete Kudos, so I can remove inappropriate or unwanted content from the platform.
+9. As an **administrator**, I want to change settings, such as creating or removing categories, adjusting expiration time of Kudos, and deleting Kudos, so I can maintain a positive and productive environment for the community.
```

Comment:
shall we merge this section into the previous one? (actors / actions)?

Reply:
User stories are distinct from the actors and the actions they can take. They provide more context, the "why" instead of just the "how". So I would prefer to keep them in.

Reply:
So, I was thinking to merge it into one section (whichever you prefer), not to simply delete one section. Now it really looks like a duplicated content.

Reply:
It's not duplicative, actors/actions is an inclusive list of requirements. The User stories are examples of why users would want to do any of those actions, which helps ensure that when the feature is developed it allows the person to accomplish their goal, rather than saying "technically the feature is done" even if it's not helpful.

Reply:
OK